docs/import_process.md

```diff
@@ -0,0 +1,79 @@
+
+The item is processed for any categories, if that feature is enabled. See [Categories import](docs/categories_import.md).
+
+The item is cleaned. This:
+
+- Cleans various fields, including:
+    - trimming whitespace from text fields
+    - validating dates
+    - formatting slugs
+    - SEO-related fields (see also [Yoast Plugin documentation](docs/yoast.md))
+- Pre-filters the body field, to clean up WordPress quirks of the HTML. Pre-filters are a series of cumulative functions. These are configurable. See [Pre-filters](docs/prefilters.md).
+- Converts the body to a series of StreamField blocks. The mapping of HTML to StreamField blocks is configurable. See [Block Builder](docs/blockbuilder.md).
+    - This stage also includes importing images from the source site, and modifying HTML `img` tags to link to the imported Image instance instead.
+- Converts any `wp:post_meta` tag nodes to a dict, which we can query later using Django's JSONField.
+
+### Update and save the page
+
+The package assumes that your own target `Page` model has an `.import_wordpress_data` function that maps the cleaned data to the available fields.
+
+The page is saved, and published according to the WordPress live/draft status. The page ID is added to a list for later.
+
+### Cache tags and items for later
+
+[Import Hooks Documentation](docs/import_hooks.md)
+
+If the XML node is configured in settings for cacheing until later, it will be stored in `WordpressImporter.tags_cache` (for top-level tags such as Authors), or `.items_cache`. Nodes that are suitable for this are those such as references to objects elsewhere in the XML, for example a header image, or an author link. We can't create the relationship until both objects have been imported.
+
+The cache will use a key, e.g, `"wp:author"`, to store a list of unique author entities found.
+
+Your settings will also have specified a function to be run after walking the XML, see [Run registered functions](#run-registered-functions).
+
+## After walking the XML
+
+Once all nodes are processed, and either imported as `Page` objects, or cached for later, some processing involving database relations is possible.
+
+### Connect rich text page links
+
+If page body StreamField contains RichText blocks with HTML `a` elements linking to relative URLs, the importer attempts to convert these to dynamic Wagtail references to `Page` objects.
+
+### Run registered functions
+
+These are configurable functions to process cached tags and items, and create the ForeignKey relationships they represent. If an XML node has been configured for cacheing until later, a corresponding function will have been registered to process the cached data.
+
+For the author example previously, this function would handle creating an Author instance for each entry found in the cache, and then querying the imported pages, and linking them to this author. This query uses the list of page IDs stored by the importer class.
```

Comment:
```suggestion
For the author example previously, this function would handle creating an Author instance for entries found in the cache, and then querying the imported pages, and linking them to this author. Only author entries that have a relation to an imported page are saved, if the author already exists the existing record is linked. This query uses the list of page IDs stored by the importer class.
```

Reply:
I agree that this would be the desired behaviour, so I've updated this, but surely the exact behaviour depends on the user's registered function?

Reply:
Thanks Nick that's a valid point and you are right. I'd be happy if you revert my suggestion.

Reply:
Okay, I've updated this again and made it clearer that the functions are for users to write.